### DISCUSSION TITLE
How can I use a different database driver (e.g. libsql for Turso)

### DISCUSSION THREAD

**Discussion by caramellok** [adonisjs/core#4299] [Help]
I know that Lucid uses Knex, and one can use the libsql wrapper with Knex [like so](https://github.com/libsql/libsql-node-sqlite3), but I can't for the life of me figure out how I'd do this in AdonisJS.

Have any AdonisJS users successfully deployed their databases to Turso?

**Comment by neupauer** [comment]
Hi! For those looking for how to connect to the TURSO database using AdonisJS (^6), here are the steps to configure it:

1. Set the connection type to `sqlite`.  
2. Use the `libsql` client.  
3. Assign the `connection.filename` to your TURSO database URL in this format: `https://{database}-{username}.{region}.turso.io?authToken={authToken}`

Additionally, make sure to install the `@libsql/sqlite3` npm package for this configuration to work.  

```base
$ npm install @libsql/sqlite3 --save
```

```js
// database.ts
const dbConfig = defineConfig({
  connection: 'sqlite',
  connections: {
    sqlite: {
      client: 'libsql',
      connection: {
        filename: 'https://{database}-{username}.turso.io?authToken={authToken}', // e.g.: https://mydb-myusername.aws-eu-west-3.turso.io?authToken=12345
      },
      useNullAsDefault: true,
      migrations: {
        naturalSort: true,
        paths: ['database/migrations'],
      },
    },
  },
})
```

Hope this helps!

**Reply by pixelpaulaus** [reply]
would it be possible to get this working in adonis v4?
